source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,931 @@
+                {
+                    case kStatus_Success: /* there is a frame.  process it */
+
+                        if( length )
+                        {
+                            prvProcessFrame( length );
+                        }
+
+                        break;
+
+                    case kStatus_ENET_RxFrameEmpty: /* Received an empty frame.  Ignore it */
+                        receiving = pdFALSE;
+                        break;
+
+                    case kStatus_ENET_RxFrameError: /* Received an error frame.  Read & drop it */
+                        PRINTF( "RX Receive Error\n" );
+                        ENET_ReadFrame( ethernetifLocal->base, &( ethernetifLocal->handle ), NULL, 0, 0, NULL );
+                        /* Not sure if a trace is required.  The MAC had an error and needed to dump bytes */
+                        break;
+
+                    default:
+                        PRINTF( "RX Receive default" );
+                        break;
+                }
+            }
+        }
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Callback for ENET interrupts. We have only enabled the Ethernet receive interrupts
+ * in the case of this driver.
+ */
+static void ethernet_callback( ENET_Type * base,
+                               enet_handle_t * handle,
+                               enet_event_t event,
+                               enet_frame_info_t * frameInfo,
+                               void * userData )
+{
+    BaseType_t needsToYield = pdFALSE;
+
+    ( void ) base;
+    ( void ) handle;
+    ( void ) frameInfo;
+    ( void ) userData;
+
+    switch( event )
+    {
+        case kENET_RxEvent:
+            vTaskNotifyGiveFromISR( receiveTaskHandle, &needsToYield );
+            portEND_SWITCHING_ISR( needsToYield );
+            break;
+
+        default:
+            FreeRTOS_printf( ( "Unknown interrupt callback %u!", event ) );
+            break;
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief This function verifies that the incoming frame needs processing.
+ *        If the frame is deemed to be appropriate, then the frame is sent to the
+ *        TCP stack for further processing.
+ * @param[in] length: The length of the incoming frame. This length should be read
+ *                    using a call to ENET_GetRxFrameSize.
+ */
+static void prvProcessFrame( int length )
+{
+    NetworkBufferDescriptor_t * pxBufferDescriptor = pxGetNetworkBufferWithDescriptor( length, 0 );
+
+    if( pxBufferDescriptor != NULL )
+    {
+        ENET_ReadFrame( ethernetifLocal->base, &( ethernetifLocal->handle ), pxBufferDescriptor->pucEthernetBuffer, length, 0, NULL );
+        pxBufferDescriptor->xDataLength = length;
+        pxBufferDescriptor->pxInterface = pxMyInterface;
+        pxBufferDescriptor->pxEndPoint = FreeRTOS_MatchingEndpoint( pxMyInterface, pxBufferDescriptor->pucEthernetBuffer );
```

Comment:
Should we drop the packet (release the buffer) if we cant find a proper end point?

Reply:
updated